quic/s2n-quic-core/src/crypto/tls.rs

```diff
@@ -23,6 +24,27 @@ pub mod slow_tls;
 #[cfg(feature = "std")]
 pub mod offload;
 
+/// Holds connection address information for establishing a TLS session.
+/// This includes both the local and remote addresses.
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
+pub struct ConnectionInfo {
+    /// The local address of the connection
+    pub local_address: SocketAddr,
+    /// The remote address of the connection
+    pub remote_address: SocketAddr,
+}
+
+impl ConnectionInfo {
+    /// Create a new ConnectionInfo
+    pub fn new(local_address: LocalAddress, remote_address: RemoteAddress) -> Self {
```

Comment:
you can add `#[doc(hidden)]` to this, since its not something we'd want external users to use